build.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+build_all_and_exit () {
+  npm run force-build
+  npm run test-base
+  npm run test-base-ws
+  exit
+}
+
+### CHECK IF THIS IS A PR ###
+if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then
+  echo "not a PR, will build everything"
+  build_all_and_exit
+fi
+
+##### DETECT CHANGES #####
+rest_pattern='ts\/src\/([A-Za-z0-9_-]+).ts' # \w not working for some reason
+ws_pattern='ts\/src\/pro\/([A-Za-z0-9_-]+)\.ts'
+diff=$(git diff origin/master --name-only)
+
+readarray -t y <<<"$diff"
+
+REST_EXCHANGES=()
+WS_EXCHANGES=()
+test_all=false
```

Comment:
should we name this variable "transpile_all" instead of "test_all"?

Reply:
"bases" might give the idea of base-tests no?

Reply:
and that's also I am about - it runs only base tests (not derived class tests) when that variable is true.

Reply:
> But that is not what this variable does. It literally tests and builds everything (base and derived tests) like we do currently

wait, how does it test everything? do i miss smth - when `test_all` is true, it just runs `build_all_and_test_base` function, which  consists of:
```
  npm run force-build
  npm run test-base
  npm run test-base-ws
  exit
```

other tests (for **derived classes**) are run from tests-manager.sh

Reply:
@carlosmiei yeah sorry you're right